### PR TITLE
Home: Dynamically highlight download link for the user's platform

### PR DESCRIPTION
**Pull request by hoffie**
**Short description of changes**

This PR adds a minimal JavaScript & stylesheet to highlight the download link for the browser-detected visitor's platform.

**Context: Fixes an issue? Related issues**

Hopefully improves usability.

**Status of this Pull Request**

**What is missing until this pull request can be merged?**

- [x] Windows testing
- [x] Mac testing
- [x] Debian/Ubuntu testing
- [x] Other platforms testing (done: Arch)

**Does this need translation?**
No.

## Checklist

- [x] I've verified that this Pull Request follows the [general code principles](https://github.com/jamulussoftware/jamulus/blob/master/CONTRIBUTING.md#jamulus-projectsource-code-general-principles)
- [x] I waited some time after this Pull Request was opened and all GitHub checks completed without errors. 
- [x] I'm sure that this Pull Request goes to the correct branch: pretty much

### CODE REVIEW

**Comment by ann0see** [comment]
Todo for self:
- [x] uploaded .zip file to Webserver to see iOS, Windows, macOS
- [x] Found occurrences of download from sf and changed them to something else --> opened Issue

What about the upgrade page? I think it would benefit from an automated redirect to the correct OS download page

**Comment by ann0see** [comment]
Doesn't work for me on Debian. What about: https://stackoverflow.com/questions/11219582/how-to-detect-my-browser-version-and-operating-system-using-javascript

Mac doesn't seem to work either.

**Review by ann0see — CHANGES_REQUESTED** [review]
As noted above, there seem to be incompatibilities...

**Comment by hoffie** [comment]
Found one (the?) issue: The script tried to modify the DOM even when it was not ready yet. I've now moved the script to the proper place. I've re-done tests on Chromium with a user agent switcher. Also rebased now.

**Review by ann0see — COMMENTED** [review]
Debian now shows "other platforms" on Firefox ESR in bold text. Not sure if that's what we want?

**Comment by hoffie** [comment]
> Debian now shows "other platforms" on Firefox ESR in bold text. Not sure if that's what we want?

Hrm, after some further research... Firefox on Ubuntu seems to identify itself as Ubuntu-based, while Firefox on Debian doesn't.
We could match all "Linux" user agents, but we would have to blacklist Android, for example.

I think I'd either keep it as it is right now in this PR (Debian points to "other platforms", which is not that wrong either, as it forwards to the full list) or maybe disable it for "Linux" completely (no highlight).

**Comment by ann0see** [comment]
I'd disable it for Linux entirely then.

File: assets/js/os-style.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+function addOperatingsystemStyle() {
+  var os = 'other';
```
Comment (ann0see) [inline assets/js/os-style.js]: Can you put that into a function getOperatingSystem() returning the OS? This makes it easier to have a redirect on jamulus.io/upgrade or jamulus.io/download if we add it one day.

**Comment by ann0see** [comment]
Although unrelated: I probably need to look into the new js loading behaviour of modern browsers again (--> async and h2 prefetching). 

Will test tomorrow ;-).

**Comment by hoffie** [comment]
> I'd disable it for Linux entirely then.

Done and reworked again.

File: assets/js/os-style.js (hunk 1 above)
Comment (hoffie) [reply assets/js/os-style.js]: Done, although it'll need rethinking/moving in that case anyway.

**Comment by pljones** [comment]
Can you post screenshots from a few platforms (ideally the native browser for that platform _on_ that platform... but fake it through the User-Agent if you must :) ).

**Comment by gilgongo** [comment]
Just tried to have a look at this locally (on Ubuntu 22.04) but get:

```
~/jamulussoftware.github.io$ bundle exec jekyll serve --trace
Could not find jekyll-4.1.1 in any of the sources
Run `bundle install` to install missing gems.
~/jamulussoftware.github.io$ bundle install
Fetching gem metadata from https://rubygems.org/
Fetching gem metadata from https://rubygems.org/.........
listen-3.2.1 requires ruby version >= 2.2.7, ~> 2.2, which is incompatible with the current version, ruby 3.0.2p107
```

Is the website no longer working with Ruby3 with this PR? I can build OK on current version (albeit you have to `bundle add webrick`)

**Comment by ann0see** [comment]
iOS works as expected.

**Comment by ann0see** [comment]
Faked Windows:

grafik

Faked macOS:

grafik

Faked Ubuntu:

grafik

Real Firefox ESR on Debian:

grafik

iOS and Android were also faked but correctly underline the "other platforms" link

**Review by ann0see — APPROVED** [review]
I think it's ok like that.

**Comment by pljones** [comment]
So long as it doesn't make things worse, it's okay, so yep.

**Review by pljones — APPROVED** [review]

**Pull request merged**